### PR TITLE
[pkg/stanza] Fix - Operators with DropOnErrorQuiet should drop log entries on error

### PR DESCRIPTION
**Pull request by SamerJ**
#### Description
Fixed issue in which Operators with DropOnErrorQuiet are not dropping log entries on error.

Note:   This issue was introduced by a bug fix meant to ensure Silent Operators are not logging errors (#35010). With this fix,
this side effect bug has been resolved.

#### Link to tracking issue
Fixes 35010

#### Testing
Added UT

### CODE REVIEW

**Comment by SamerJ** [comment]
@fatsheep9146 
Let me know why you think. 
Also, should there be any questions, let me know 😄  .

File: pkg/stanza/operator/helper/transformer.go [hunk 1]
```diff
@@ -106,10 +106,6 @@ func (t *TransformerOperator) HandleEntryError(ctx context.Context, entry *entry
 		}
 	}
 
-	if t.OnError == SendOnErrorQuiet || t.OnError == DropOnErrorQuiet {
-		return nil
-	}
```
Comment (djaglowski) [inline pkg/stanza/operator/helper/transformer.go:111]: Why is this removed?

Comment (SamerJ) [reply pkg/stanza/operator/helper/transformer.go:111]: The error returned by this method isn't soley used to log a message. Sometimes the error is read and based on that we decide if a log entry should be dropped or not.
A concrete example would be https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.112.0/pkg/stanza/operator/helper/parser.go#L111

I was fearful of any refactoring because I don't want to go against a possible design target.

Comment (djaglowski) [reply pkg/stanza/operator/helper/transformer.go:111]: Won't think change result in a problem for transformers that are not parsers?

Comment (SamerJ) [reply pkg/stanza/operator/helper/transformer.go:111]: I was not too fearful of removing this block, because it didn't exist before my previous PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35010
That being said, I see your point.
My rational is as such:
We can't replace an `error` with `nil` at this level as that would make it impossible for the caller to know if they should drop the entry or send it.
The caller has visibility on the config like `OnError`, so I ended moving the responsibility to the caller.

Let me know what you think.

**Review by djaglowski — APPROVED** [review]
